src/js/component/App.js

```diff
@@ -34,33 +38,19 @@ class App {
 
   initialContent() {
     this.sources.forEach((item)=>{
-      this.nav.innerHTML +=
-        `<li><a data-load-channel="${item.inst}">${item.source}</a></li>`
-
-      this.main.innerHTML +=
-        `<div class="thumbnail">
-          <div class="caption">
-          <h3>${item.label}</h3>
-          <p><a href="#" class="btn btn-primary" data-load-channel="${item.inst}" role="button">${item.source}</a></p>
-          </div>
-        </div>`   
+      this.nav.innerHTML += view.navHTML(item)
+      this.main.innerHTML += view.mainHTML(item)
     }); 
   }  
 
-  async get(source) {
+  get(source) {
     this.spinner.enableSpinner()    
```

Comment:
spinner also whould be delegated to the view

Reply:
Delegated as a separate Class inside the Views